zebra-state/src/service.rs

```diff
@@ -209,6 +202,21 @@ impl StateService {
             for (child, rsp_tx) in queued_children {
                 let child_hash = child.hash;
                 tracing::trace!(?child_hash, "validating queued child");
+
+                // required by validate_and_commit
+                let mandatory_checkpoint = self.network.mandatory_checkpoint_height();
+                if child.height <= mandatory_checkpoint {
+                    panic!(
+                        "invalid non-finalized block height: the canopy checkpoint is mandatory, pre-canopy blocks, and the canopy activation block, must be committed to the state as finalized blocks"
+                    );
+                }
+
+                // required by check_contextual_validity
+                let relevant_chain =
+                    self.any_ancestor_blocks(child.block.header.previous_block_hash);
+                assert!(relevant_chain.len() >= POW_AVERAGING_WINDOW + POW_MEDIAN_BLOCK_SPAN,
+                "contextual validation requires at least 28 (POW_AVERAGING_WINDOW + POW_MEDIAN_BLOCK_SPAN) blocks");
```

Comment:
Nit: Splitting the line to make it easier to read.

```suggestion
                assert!(
                    relevant_chain.len() >= POW_AVERAGING_WINDOW + POW_MEDIAN_BLOCK_SPAN,
                    "contextual validation requires at least \
                    28 (POW_AVERAGING_WINDOW + POW_MEDIAN_BLOCK_SPAN) blocks"
                );
```

Reply:
Fixed in ca02edac.